net/src/lib.rs

```diff
@@ -218,6 +223,7 @@ impl<P: StoreParams> NetworkService<P> {
     pub fn get(&self, cid: Cid) -> GetQuery<P> {
```

Comment:
Isn't get the one that goes against the local db, and fetch the one that might use the network?

Also, it seems a bit brute force to just wake the swarm in any case. E.g. if you fetch, and the data is local, there is no need to wake. Ideally the waking would automatically happen as soon as you produce work. But I guess this will do for now.